store/commitment/types/batch.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+// KVPair is a key-value pair. It is used to represent a change in a Batch.
+// NOTE: The Value can be nil, which means the key is deleted.
+type KVPair struct {
+	Key   []byte
+	Value []byte
+}
+
+// Batch is a change set that can be committed to the database atomically.
+type Batch struct {
+	Pairs []KVPair
+}
+
+// NewBatch creates a new Batch instance.
+func NewBatch() *Batch {
+	return &Batch{
+		Pairs: []KVPair{},
```

Comment:
nit: can we make this size zero and maybe add a `Grow` method since we should know the length of pairs ahead of time